sdk/cosmos/azure-cosmos/azure/cosmos/documents.py

```diff
@@ -94,197 +104,147 @@ class ConsistencyLevel(object):
-    """Specifies the kind of partitioning to be applied.
-
-    :ivar str PartitionKind.Hash:
-        The partition key definition path is hashed.
+    Session: Literal["Session"] = "Session"
+    """Session Consistency guarantees monotonic reads (you never read old data,
+    then new, then old again), monotonic writes (writes are ordered) and
+    read your writes (your writes are immediately visible to your reads)
+    within any single session.
     """
-
-    Hash = "Hash"
-    MultiHash = "MultiHash"
-
-
-class DataType(object):
-    """Specifies the data type of index specs.
-
-    :ivar str Number:
-        Represents a numeric data type.
-    :ivar str String:
-        Represents a string data type.
-    :ivar str Point:
-        Represents a point data type.
-    :ivar str LineString:
-        Represents a line string data type.
-    :ivar str Polygon:
-        Represents a polygon data type.
-    :ivar str MultiPolygon:
-        Represents a multi-polygon data type.
+    Eventual: Literal["Eventual"] = "Eventual"
+    """Eventual Consistency guarantees that reads will return a subset of
+    writes. All writes will be eventually be available for reads.
     """
-
-    Number = "Number"
-    String = "String"
-    Point = "Point"
-    LineString = "LineString"
-    Polygon = "Polygon"
-    MultiPolygon = "MultiPolygon"
-
-
-class IndexingDirective(object):
-    """Specifies whether or not the resource is to be indexed.
-
-    :ivar int Default:
-        Use any pre-defined/pre-configured defaults.
-    :ivar int Exclude:
-        Index the resource.
-    :ivar int Include:
-        Do not index the resource.
+    ConsistentPrefix: Literal["ConsistentPrefix"] = "ConsistentPrefix"
+    """ConsistentPrefix Consistency guarantees that reads will return some
+    prefix of all writes with no gaps. All writes will be eventually be
+    available for reads.
     """
 
-    Default = 0
-    Exclude = 1
-    Include = 2
 
-
-class ConnectionMode(object):
-    """Represents the connection mode to be used by the client.
-
-    :ivar int Gateway:
-        Use the Azure Cosmos gateway to route all requests. The gateway proxies
-        requests to the right data partition.
+class IndexingMode:
+    """Specifies the supported indexing modes."""
+    Consistent: Literal["consistent"] = "consistent"
+    """Index is updated synchronously with a create or update operation. With
+    consistent indexing, query behavior is the same as the default
+    consistency level for the collection.
+    The index is always kept up to date with the data.
     """
-
-    Gateway = 0
-
-
-class PermissionMode(object):
-    """Enumeration specifying applicability of a permission.
-
-    :ivar str PermissionMode.NoneMode:
-        None.
-    :ivar str PermissionMode.Read:
-        Permission applicable for read operations only.
-    :ivar str PermissionMode.All:
-        Permission applicable for all operations.
+    Lazy: Literal["lazy"] = "lazy"
```

Comment:
should this be capital Lazy to be consistent with the others?

Reply:
I'm honestly not sure why some of the constants are in lower case vs others in camel case.... though I think I'll just leave them as-is in this PR.